test/integration/testutil/assertions.go

```diff
@@ -75,3 +82,54 @@ func AssertLogsForExec(t *testing.T, logs *observer.ObservedLogs, expectedURL, e
 	// Request and response logs must have the same request ID
 	assert.Equal(t, firstLog.ContextMap()["request_id"], secondLog.ContextMap()["request_id"])
 }
+
+func assertValidAsciicast(t *testing.T, asciicast string, expectedHeader wsproxy.AsciicastHeader, expectedEvents []string) {
+	t.Helper()
+
+	lines := strings.Split(strings.TrimSpace(asciicast), "\n")
+	expectedLines := 1 + len(expectedEvents) // Include header line and events
+	require.Len(t, lines, expectedLines, "asciicast should have %d lines", expectedLines)
+
+	assertAsciicastHeader(t, lines[0], expectedHeader)
+
+	for i, event := range expectedEvents {
+		assertAsciicastEvent(t, lines[i+1], event)
+	}
+}
+
+func assertAsciicastHeader(t *testing.T, headerLine string, expectedHeader wsproxy.AsciicastHeader) {
+	t.Helper()
+
+	var header wsproxy.AsciicastHeader
+
+	err := json.Unmarshal([]byte(headerLine), &header)
+	require.NoError(t, err)
+
+	assert.Equal(t, expectedHeader.Version, header.Version)
+	assert.Equal(t, expectedHeader.Width, header.Width)
+	assert.Equal(t, expectedHeader.Height, header.Height)
+	assert.Positive(t, header.Timestamp)
+	assert.Equal(t, expectedHeader.Command, header.Command)
+	assert.Equal(t, expectedHeader.User, header.User)
+	assert.Equal(t, expectedHeader.K8sMetadata, header.K8sMetadata)
```

Comment:
I would zero the `timestamp` field and then compare `assert.Equal(t, expectedHeader, header)`. If we add/remove field from the `wsproxy.AsciicastHeader` struct, we don't need to update these assertions.